packages/flutter_tools/lib/src/commands/migrate_abandon.dart

```diff
@@ -0,0 +1,129 @@
+      valueHelp: 'path',
+    );
+    argParser.addFlag(
+      'force',
+      abbr: 'f',
+      help: 'Delete the migrate working directory without asking for confirmation.',
+    );
+  }
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  final Terminal terminal;
+
+  final MigrateUtils migrateUtils;
+
+  @override
+  final String name = 'abandon';
+
+  @override
+  final String description = 'Deletes the current active migration working directory.';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory(logger: logger, fileSystem: fileSystem);
+    final FlutterProject project = projectDirectory == null
+      ? FlutterProject.current()
+      : flutterProjectFactory.fromDirectory(fileSystem.directory(projectDirectory));
+    Directory workingDirectory = project.directory.childDirectory(kDefaultMigrateWorkingDirectoryName);
+    final String? customWorkingDirectoryPath = stringArg('staging-directory');
+    if (customWorkingDirectoryPath != null) {
+      if (fileSystem.path.isAbsolute(customWorkingDirectoryPath)) {
+        workingDirectory = fileSystem.directory(customWorkingDirectoryPath);
+      } else {
+        workingDirectory = project.directory.childDirectory(customWorkingDirectoryPath);
+      }
+      if (!workingDirectory.existsSync()) {
+        logger.printError('Provided working directory `$customWorkingDirectoryPath` '
+                          'does not exist or is not valid.');
+        return const FlutterCommandResult(ExitStatus.fail);
+      }
+    }
+    if (!workingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('flutter migrate start', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    logger.printStatus('\nAbandoning the existing migration will delete the '
+                       'migration working directory at ${workingDirectory.path}');
+    final bool force = boolArg('force') ?? false;
+    if (!force) {
+      String selection = 'y';
+      terminal.usesTerminalUi = true;
+      try {
+        selection = await terminal.promptForCharInput(
+          <String>['y', 'n'],
+          logger: logger,
+          prompt: 'Are you sure you wish to continue with abandoning? (y)es, (N)o',
+          defaultChoiceIndex: 1,
+        );
+      } on StateError catch(e) {
+        logger.printError(
+          e.message,
+          indent: 0,
+        );
+      }
+      if (selection != 'y') {
+        return const FlutterCommandResult(ExitStatus.success);
+      }
+    }
+
+    workingDirectory.deleteSync(recursive: true);
```

Comment:
Depending on the platform, deletion can be flaky, and recursive directory delete especially so. So, I'd suggest wrapping this in a `try {} catch {}` that handles `FileSystemException` and asks the user to manually delete it rather than crashing.